### PR TITLE
Compile `coqpp` inside the `bin/` folder and make it available after installation

### PR DESCRIPTION
**Pull request by ppedrot**
Compile `coqpp` inside the `bin/` folder and make it available after installation. No need to put it inside `.gitignore`.

### CODE REVIEW

**Comment by ppedrot** [comment]
I am assigning @gares manually to this PR as he is supposed to be the secondary maintainer of makefiles.

**Comment by ejgallego** [comment]
The correct procedure is to ask him for a review.

**Comment by ppedrot** [comment]
@Zimmi48 it looks like it's performing a nonsense action, namely `OCAMLC -a bin/coqpp`. I do not understand the makefile enough to grasp where something went wrong unluckily.

File: Makefile.build [hunk 1]
```diff
@@ -369,7 +369,7 @@ grammar/grammar.cma : $(GRAMCMO)
 	$(SHOW)'OCAMLC -a $@'
 	$(HIDE)$(GRAMC) $^ -linkall -a -o $@
 
-grammar/coqpp: $(COQPPCMO) grammar/coqpp_main.ml
+$(COQPP): $(COQPPCMO) grammar/coqpp_main.ml
 	$(SHOW)'OCAMLC -a $@'
 	$(HIDE)$(GRAMC) $^ -linkall -o $@
```
Comment (SkySkimmer) [inline Makefile.build]: >@Zimmi48 it looks like it's performing a nonsense action, namely OCAMLC -a bin/coqpp

It's from here. The SHOW is lying about what is done.

Comment (ppedrot) [reply Makefile.build]: It doesn't explain why it fails though. The compilation command looks legit and it does work with other CI targets.

**Comment by SkySkimmer** [comment]
`bin` is created by configure and not preserved by artefacts. Then ocaml dies when we try to output to bin/ for the automatic dependency stuff.
See https://github.com/coq/coq/commit/cd45eda96c0026fc8962ba51b3e3750be946b96f for a possible solution, otherwise we can `mkdir bin` in the script for the doc job.

**Review by ejgallego — APPROVED** [review]

**Comment by ejgallego** [comment]
@ppedrot you assigned @gares however this doesn't work for him as this PR won't appear on his Project. We discussed this in the past, no point on assigning PRs to other people as for them to be ignored.

**Comment by ejgallego** [comment]
This is kinda trivial so taking it; I'll take care of it tomorrow unless someone steps in.

**Review by gares — APPROVED** [review]

**Pull request merged**